hyperspy/io_plugins/bcf.py

```diff
@@ -946,246 +908,217 @@ def parse_hypermap(self, index=None,
         if index is None:
             index = self.def_index
         if type(cutoff_at_kV) in (int, float):
-            eds = self.header.get_spectra_metadata(index)
-            cutoff_chan = eds.energy_to_channel(cutoff_at_kV)
+            eds = self.header.spectra_data[index]
+            max_chan = eds.energy_to_channel(cutoff_at_kV)
         else:
-            cutoff_chan = None
-
+            max_chan = self.header.estimate_map_channels(index=index)
+        shape = (-(-self.header.image.height // downsample),
```

Comment:
Why are you using the negative sign in `(-(-`? If it is to avoid getting a `0` I would suggest the more explicit:

```python


shape = (self.header.image.height // downsample, self.header.image.width // downsample, max_chan)
shape = [ss if ss >= 1 else 1 for ss in shape]
```

Reply:
there is such math function: floor, what I need here is roof, that is the trick

Reply:
i.e. `5 // 2` would give 2, but I need 3, but I also need that `4 // 2` would give 2

Reply:
I could add comment that it is roof, maybe it would be clear then what `-(-` does.

Reply:
Then, maybe what you need is ``np.ceil`` or ``math.ceil``

Reply:
I think math.ceil is more approprate as it returns integer. I wonder how did I missed that simple solution...